backend/app/api/v1/endpoints/task.py

```diff
@@ -0,0 +1,372 @@
+        )
+    except Exception as e:
+        log.error(f"Failed to update task: {e}")
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Failed to update task: {str(e)}",
+        )
+
+
+@router.put("/change-status/{task_id}", response_model=TaskInDB)
+async def update_task_status(
+    task_id: int,
+    status: bool = Form(...),
+    db: AsyncSession = Depends(get_db),
+    current_user_id: int = Depends(get_current_user),
+    user_role: str = Depends(get_current_user_role),
+):
+    log.info(f"Updating status of task with id: {task_id} to {status}")
+    try:
+        db_task = await fetch_data_by_id(db, Task, task_id)
+        
+        await check_authorization_if_forbiden(current_user_id, db_task, user_role)
+
+        db_task.status = status
+        updated_task = await update_instance(db, db_task)
+        
+        log.info(f"Status of task with id {task_id} updated successfully")
+        return updated_task
+    except Exception as e:
+        log.error(f"Failed to update task status: {e}")
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Failed to update task status: {str(e)}",
+        )
+
+
+
+@router.delete("/tasks/{task_id}")
+async def delete_task(
+    task_id: int,
+    db: AsyncSession = Depends(get_db),
+    current_user_id: int = Depends(get_current_user),
+    user_role: str = Depends(get_current_user_role),
+):
+    log.info(f"Deleting task with id: {task_id}")
+    try:
+        db_task = await fetch_data_by_id(db, Task, task_id)
+        
+        await check_authorization_only_admin(db_task, user_role)
+
+        await delete_instance(db, db_task)
+        
+        return {"message": "Task deleted successfully"}
+    except Exception as e:
+        log.error(f"Failed to delete task: {e}")
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Failed to delete task: {str(e)}",
+        )
+
+
+
+@router.put("/task-delete-request/{task_id}")
+async def request_delete_task(
+    task_id: int,
+    db: AsyncSession = Depends(get_db),
+    current_user_id: int = Depends(get_current_user),
+    user_role: str = Depends(get_current_user_role),
+):
+    log.info(f"Deleting task with id: {task_id}")
+    try:
+        db_task = await fetch_data_by_id(db, Task, task_id)
+        
+        await check_authorization_if_forbiden(current_user_id, db_task, user_role)
+
+        db_task.delete_request = True
+        updated_task = await update_instance(db, db_task)
+        
+        log.info(f"Task with id {task_id} delete requested successfully")
+        return updated_task
+    except Exception as e:
+        log.error(f"Failed to request delete task: {e}")
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Failed to delete task: {str(e)}",
+        )
```

Comment:
Add a new line here.

Reply:
Added a new line, actually applied ruff on the files:
```
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail=f"{SystemMessages.ERROR_FAILED_TO_REQUEST_DELETE_TASK} {str(e)}",
        )
 
```